MultiThreading.Task5.Threads.SharedCollection/Program.cs

```diff
@@ -5,11 +5,21 @@
  * Use Thread, ThreadPool or Task classes for thread creation and any kind of synchronization constructions.
  */
 using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
 
 namespace MultiThreading.Task5.Threads.SharedCollection;
 
 class Program
 {
+    private const int CollectionLength = 10;
+    private static readonly ManualResetEventSlim _producerAddedNewValue = new(false);
+    private static readonly ManualResetEventSlim _producerFinishedWork = new(false);
+    private static readonly ManualResetEventSlim _consumerHandledValue = new(false);
+    private static readonly List<int> _collection = new();
+    private static readonly object lockObject = new();
```

Comment:
Suppose we don't need lock at all? I removed it and nothing has changed

Reply:
Yes, you are right, current synchronization mechanism isolates enough our collection.